drivers/led/issi/is31fl3729.c

```diff
@@ -0,0 +1,219 @@
+
+    // Set Spread Spectrum Register if applicable
+    is31fl3729_write_register(addr, IS31FL3729_REG_SPREAD_SPECTRUM, ((IS31FL3729_SPREAD_SPECTRUM & 0b1) << 4) | ((IS31FL3729_SPREAD_SPECTRUM_RANGE & 0b11) << 2) | (IS31FL3729_SPREAD_SPECTRUM_CYCLE_TIME & 0b11));
+
+    // Set PWM Frequency Register if applicable
+    is31fl3729_write_register(addr, IS31FL3729_REG_PWM_FREQUENCY, IS31FL3729_PWM_FREQUENCY);
+
+    // Set Golbal Current Control Register
+    is31fl3729_write_register(addr, IS31FL3729_REG_GLOBAL_CURRENT, IS31FL3729_GLOBAL_CURRENT);
+
+    // Set to Normal operation
+    is31fl3729_write_register(addr, IS31FL3729_REG_CONFIGURATION, IS31FL3729_CONFIGURATION);
+
+    // Wait 10ms to ensure the device has woken up.
+    wait_ms(10);
+}
+
+void is31fl3729_set_color(int index, uint8_t red, uint8_t green, uint8_t blue) {
+    is31fl3729_led_t led;
+    if (index >= 0 && index < IS31FL3729_LED_COUNT) {
+        memcpy_P(&led, (&g_is31fl3729_leds[index]), sizeof(led));
+
+        if (g_pwm_buffer[led.driver][led.r] == red && g_pwm_buffer[led.driver][led.g] == green && g_pwm_buffer[led.driver][led.b] == blue) {
+            return;
+        }
+
+        g_pwm_buffer_update_required[led.driver] = true;
+        g_pwm_buffer[led.driver][led.r]          = red;
+        g_pwm_buffer[led.driver][led.g]          = green;
+        g_pwm_buffer[led.driver][led.b]          = blue;
+    }
+}
+
+void is31fl3729_set_color_all(uint8_t red, uint8_t green, uint8_t blue) {
+    for (int i = 0; i < IS31FL3729_LED_COUNT; i++) {
+        is31fl3729_set_color(i, red, green, blue);
+    }
+}
+
+void is31fl3729_set_scaling_register(uint8_t index, uint8_t red, uint8_t green, uint8_t blue) {
+    is31fl3729_led_t led;
+    memcpy_P(&led, (&g_is31fl3729_leds[index]), sizeof(led));
+
+    // need to do a bit of checking here since 3729 scaling is per CS pin.
+    // not the usual per RGB key as per other ISSI drivers
+    // only enable them, since they should be default disabled
+    int cs_red   = (led.r & 0x0F) - 1;
+    int cs_green = (led.g & 0x0F) - 1;
+    int cs_blue  = (led.b & 0x0F) - 1;
+
+    g_scaling_registers[led.driver][cs_red]   = red;
+    g_scaling_registers[led.driver][cs_green] = green;
+    g_scaling_registers[led.driver][cs_blue]  = blue;
+
+    g_scaling_registers_update_required[led.driver] = true;
+}
+
+void is31fl3729_update_pwm_buffers(uint8_t addr, uint8_t index) {
+    if (g_pwm_buffer_update_required[index]) {
+        is31fl3729_write_pwm_buffer(addr, index);
+
+        g_pwm_buffer_update_required[index] = false;
+    }
+}
+
+void is31fl3729_update_scaling_registers(uint8_t addr, uint8_t index) {
+    if (g_scaling_registers_update_required[index]) {
+        for (uint8_t i = 0; i < IS31FL3729_SCALING_REGISTER_COUNT; i++) {
+            is31fl3729_write_register(addr, IS31FL3729_REG_SCALING + i, g_scaling_registers[index][i]);
+        }
+
+        g_scaling_registers_update_required[index] = false;
```

Comment:
```suggestion
    if (driver_buffers[index].scaling_buffer_dirty) {
        for (uint8_t i = 0; i < IS31FL3729_SCALING_REGISTER_COUNT; i++) {
            is31fl3729_write_register(addr, IS31FL3729_REG_SCALING + i, driver_buffers[index].scaling_buffer[i]);
        }

        driver_buffers[index].scaling_buffer_dirty = false;
```

Reply:
Thanks, I've just push a new commit to resolved this.